### ISSUE TITLE
Beta10: Manage Stations page

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#994]
Some of these are more important than others but since this is such a big change in how people will interact with their stations and receivers, the fewer loose ends and unpolished bits we can leave the smoother the transition will be for everyone.

- [ ] Need option to copy properties previous deployment when adding a deployment to a station. Similar to the existing feature when creating a deployment except that instead of copying the most recent properties for that particular receiver, it will copy the most recent properties for that station, regardless of what receiver was there previously.

- [x]  Move the "Add new station" button to _above_ the filter/search bar so that those controls are adjacent to the top of map and table

- [x] Filter station by ID only returns exact match. It should return partial match as well (filter by name _does_ return partial matches.)

- [x] Filter by receiver ID only returns the station where that receiver is actively deployed (if there is one) rather than all stations that receiver has been associated with. 

- [x] Toggling exclude/include active stations only works with an active filter and not the default list

  - [x] The default state should be toggled be "exclude inactive stations"

- [x] Station status isn't populated and there is no option in the dropdown. 

- [ ] In the receiver drop-down in the "add deployment" modal... A filterable list of all receivers registered to the project, along with an indication of whether they are available or not (have an active deployment somewhere else). Can you style within a dropdown? Maybe a small sized table? At the very least for now, only receivers not currently deployed at a station should be displayed. 

I think we will still need some sort of receiver summary, which shows a list of all receivers registered to a project, as well as a link to an overview of a each receiver's history and status, including most recent data from that receiver (may not be associated with a deployment), the Receiver Timeline, a list of all stations the receiver has been associated with. Maybe ultimately such a table could go in the Add Deployment modal, or maybe its own page...

- [ ] On the Manage Receivers page currently, there are helpful links to a couple overview plots (by deployment and by receiver). Both of these are really useful tools and we want to preserve easy access to something like them. 

- [ ] Admin tool to delete receiver deployments not yet present on the new layout. Maybe that should go on the Edit Deployment page?

- [ ] In modify this station, the landowner list should be filterable or selectable upon typing. 

- [x] Some clarification in the right hand side that what we're seeing are receiver deployments associated with a station.
  - [x] Header above that entire lower section like "Receiver deployments associated with this station" or something similar.

- [x] In the receiver deployment history, swap "Height" field for "Dongle type.

- [x] Stations map under Manage Stations produces 'close' popup which doesn't disappear each time a station is hovered over. Maybe the labels should just disappear when the mouse is not over them, rather than having to be closed explicitly.

- [x] Some stations without lat/lon show up on null island. Not sure how those should be presented... Maybe a line below the map indicating that X station lacked valid location information so aren't displayed? Doesn't matter in the table view.

A few general comments...

I'm not not really clear on how receiver deployments are managed. It will be easy to change properties on an active station: End the existing receiver deployment, copy the details from the previous, and change anything that needs to be changed, then save. But I'm not sure how to change anything in the past. The station cannot be changed, but the receiver ID can be. Maybe we want that the other way around? 

Classifying stations by site will work in most cases. But there will be exceptions that people will have to track down and resolve. I'm not sure how easy that will be. I've noticed a couple in the Project 1 but haven't dug in to see what the deal is.

How are we going to deal with Tests deployments?

Comments over time are still associated with individual receiver deployments instead of the station (this isn't actually a web issue).

How to retain important information in the tables now that we have less for columns... We have examples already of multi-line table rows (Manage Receivers often has 4 lines per row with the link on the right hand side). This isn't really an issue, and if we are fine with multi-line rows in the new version it helps get around the limitation of having less horizontal space, so maybe this becomes a non-issue.

**Comment by pmorrill** [comment]
I have checked many of the above items, indicating that they have changed to match your requests here. You need to check them over. Those that are still unchecked require some further discussion, or just have not been dealt with yet. Will add comments on a couple here:

- _filter station by id requires exact match;_ the id field is an integer, so a partial match is not very practical - I am not even sure if T-SQL supports a 'LIKE' clause for numeric values.....
- _In the receiver drop-down in the "add deployment" modal..._; This list **should** only show those not already in active deployments on the project; Double check this and let me know of a specific example maybe....
- _In modify this station, the landowner list should be filterable or selectable upon typing_; if I start typing, I do jump to reasonable match (browser defines reasonable). We can make this more sophisticated of course: but is it critical (ie: to hold up deployment to live)?

**Comment by pmorrill** [comment]
Regarding the ability to copy over an existing deployment, and start a new one, I see a couple of options that you should consider:

Put the functionality on the 'Edit Deployment' page, rather than the stations page. This would mean a tool at the top of the page with clear explanation and offering certain options such as:

- change the receiver on this active deployment (terminating the current one);
- re-activate this terminated deployment (would actually copy and create a new deployment with all fields the same)

We could also offer similar options in a decision popup that would appear when you click 'Modify this deployment', but I lean towards the above, for a clean presentation.

**Comment by pmorrill** [comment]
_Some stations without lat/lon show up on null island. Not sure how those should be presented... Maybe a line below the map indicating that X station lacked valid location information so aren't displayed? Doesn't matter in the table view._

Can you give specific example? I am not finding a project that shows a bad point on 'null island' whatever that is.

**Comment by jsayers-bsc** [comment]
- That should have read _receiverID_ by partial match. That works now, and also returns previous stations. Perfect.

![image](https://user-images.githubusercontent.com/55001144/126660171-407c8411-d8aa-4b1c-9c4e-4e43498756d7.png)

- It currently shows an extensive list. I'm not entirely sure what the status is of each of these (related to a comment about still needing some sort of "all receivers" overview page) but the current list of undeployed receivers for project 1 is just 19. https://motus.org/data/project/recvs?projectID=1#tabs and the beta version shows many more.
- No not critical

**Comment by pmorrill** [comment]
Cancel my request re null island. I have simulated what I need. And, discovered null island! I never realized there is an actual tiny island at 0,0. Cool.

**Comment by jsayers-bsc** [comment]
> Regarding the ability to copy over an existing deployment, and start a new one, I see a couple of options that you should consider:
> 
> Put the functionality on the 'Edit Deployment' page, rather than the stations page. This would mean a tool at the top of the page with clear explanation and offering certain options such as:
> 
>     * change the receiver on this active deployment (terminating the current one);
> 
>     * re-activate this terminated deployment (would actually copy and create a new deployment with all fields the same)
> 
> 
> We could also offer similar options in a decision popup that would appear when you click 'Modify this deployment', but I lean towards the above, for a clean presentation.

Right now in beta to make changes to an active station's configuration you must first terminate the existing receiver deployment associated with it, then the "Add Deployment" option becomes visible. 

That brings up a modal where a receiver can be selected, which then leads to the typical receiver deployment page (with the receiver and station pre-populated).

I tend to agree with you that all of this might be better on the Edit Deployment page. 

- Put the dropdown to select receivers there instead of the modal
  - The idea of changing the receiver for an active deployment here was something I originally pitched, but thought it was decided that that introduced too much complexity in the back end with determining end times etc. That's fine since ending a deployment and creating a new one is what people are used to anyway
- This is where the "Copy information from the previous deployment" already lived anyway so copying the _station's_ most recent deployment makes sense here

**Comment by pmorrill** [comment]
I think that putting the recv selection drop-down on the edit deployments page is reasonable, but recv has to be selected **before** the deployment is created. So, the page would display that option ONLY, without the rest of the edit fields. Once the recv form was submitted, you'd move directly to the field editing. If the operation was a copy/clone of an existing deployment, several fields, and all antenna / node data, would be already pre-filled.

**Comment by Hussell** [comment]
Layout issues on smaller screens:
![Screenshot from 2021-07-22 13-50-56](https://user-images.githubusercontent.com/28784000/126685802-ef6fe999-c1ab-41a0-a2cf-c60a17c3ae86.png)

This is just on a desktop monitor, not a tablet or phone.

Also, is there a way to link from elsewhere directly to the page to edit a specific station?

**Comment by Hussell** [comment]
Remember to check the layout in other languages:

![Screenshot from 2021-07-22 14-08-46](https://user-images.githubusercontent.com/28784000/126687845-28f8a209-048b-47e7-bc14-a477a1605728.png)

**Comment by pmorrill** [comment]
> Also, is there a way to link from elsewhere directly to the page to edit a specific station?

Yes. By example:

https://beta.motus.org/data/project/stations?projectID=1&setStationId=197

The setStationId param triggers a redirect to dsplay the summary / manage tools

**Comment by jsayers-bsc** [comment]
> 
> 
> > Also, is there a way to link from elsewhere directly to the page to edit a specific station?
> 
> Yes. By example:
> 
> https://beta.motus.org/data/project/stations?projectID=1&setStationId=197
> 
> The setStationId param triggers a redirect to dsplay the summary / manage tools

Excellent. Direct links are super useful.

**Comment by jsayers-bsc** [comment]
This is looking great, Paul. 

There are a few things I've noticed....

- Port dropdown is empty
- Receiver ID in station overview has "- 0" appended
- Angle and Polarization sections can be moved to the "Advanced Antenna Properties" tab'
- Navigating back from a previous configuration page doesn't always go back to proper station
- Create new Configuration seems like it should only be available from the most recent/active deployment? Or maybe from the station overview? 
- Terminating a deployment in the current way doesn't allow notes to be added (e.g. "ending this deployment as a test"). That may not be an issue once we figure out how we're consolidating notes from individual deployments at the station level...
- Copy Configuration Details doesn't populate the receiver ID above it
![image](https://user-images.githubusercontent.com/55001144/128856652-a6aeb65e-3822-4b77-9338-ef774fcce31c.png)
- The receiver just freed up at this station is not present in the dropdown in this particular case
- Some way of finding or filtering from the receiver list would be very helpful. Is a filterable table a viable option in this location?

**Comment by pmorrill** [comment]
Have fixed a few items:

- removed the '- 0' from receiver id
- moved angle and polarization fields to advanced tab

I also added a link to the recv timeline - but it likely needs some adjustment.

Comments:

The port dropdown only gets populated once you select a dongle type; same with the frequency dropdown.

The new config tool is designed to force you to select the receiver that you want to use, first. The copy config. option is secondary and layered on top - it does not reset the recv selection above it, by design. It's a cart / horse thing. But there may be a better way to lay it out.

Can you provide a test case where navigating back from the recv deployment page does not take you to the correct station summary? It seems to work ok for me, but we need to track it down. Are you using the breadcumbs? Or the 'Save changes and exit' button? Or the 'Cancel' button?

**Comment by leberrigan** [comment]
@pmorril - There is a bug on **modify station configuration** page where the 'view antenna map' button no longer works. Javascript error: 
```
Uncaught TypeError: document.querySelector(...) is null
    displayPreviewMap https://sandbox.motus.org/data/station/deployment/edit?projectID=78&id=6197&edit=1:1516
```

**Comment by jsayers-bsc** [comment]
@pmorrill Lucas and I are going to be chatting about this tomorrow (Tuesday) and will then post our suggestions here.

One quick question: how restricted (or not) is the modal format for presenting data? Is a filterable/searchable table similar to what currently exists on the "manage receivers" page possible in this format?

**Comment by pmorrill** [comment]
I think that if we need to go to a more elaborate search mechanism to select a recv, we might replace the modal with a standard page.

**Comment by jsayers-bsc** [comment]
> I think that if we need to go to a more elaborate search mechanism to select a recv, we might replace the modal with a standard page.

We definitely need a receiver list with more options to find and select receivers. We actually already have something that ticks nearly all the boxes (we'd probably want to edit the fields a bit): https://motus.org/data/projectReceivers?n=1000&q=&f=1&o=0a1a2a3a&id=1

We'd want to incorporate something with this functionality into the receiver selection, whether modifying a station with an existing active deployment or adding a receiver to a station with no active deployment.

**Comment by jsayers-bsc** [comment]
This relates only to the modal/page to allow easy changes to the receiver or antennas (creating a new deployment in the process)

The "Modify configuration" page (or modal) could have these 4 options. If there is no currently active deployment, the first option wouldn't be present but the rest would be mostly the same.

1) Date and time to terminate the existing configuration. Either prepopulated with the current UTC time or with a button beside it to do that. 

2) Date and time when new configuration becomes active. Along with a check box to "begin new configuration when previous one ended" which is defaulted to yes (so this box auto-populates with the same unless the user selects otherwise)

3) Check box to "Copy RECEIVER from most recent station configuration". Defaulted to yes?

4) Check box to "Copy ANTENNA PROPERTIES from most recent station configuration". Defaulted to yes?

- If Copy Receiver from most recent configuration is _not_ checked, you will then be directed to the "receiver list" page (similar to the existing one in the explore side) that will allow a receiver to be selected. Question is whether this should be a redirect to a new page, or if the 4 options above should be presented, along with an embedded receiver list if applicable on its own page?

**Comment by ajmacdonald3** [comment]
We need to make the difference between creating a new configuration and fixing an error in the current configuration very clear because this has implications for whether the metadata is correct. I suggested something along the lines of the following for the button text:

- Create new configuration/deployment
- Fix error in current configuration/deployment

We also discussed having an extra step to get to "Fix error in current configuration", and having "View current configuration" on the first page instead, which then leads to a page with all the current properties where there is the option to "Fix error in current configuration".

**Comment by StuMackenzie** [comment]
**What are we going to lose in Manage Receivers Tool ?**

**1) List of receivers** - I think one idea may be to just to add a list of 'receivers' registered to each project to the current tabs - 
Tabs = **List of Stations | Map of Stations | List of Receivers (eg. current Manage Receivers Page) | Station Status Map ** 

List of Receivers should include: Receiver ID (link to receiver timeline), Status (active, undeployed, terminated, etc), Latest Deployment,???  @jsayers-bsc populate table.

**2) Map of receiver status** - eg. - https://motus.org/data/project/receivers/map?id=1 

a) @Hussell - Can this map be modified to be based on stations instead of receivers?
  
 If yes, then for now just 'simply' add a tab "Station Status Map"

**3) Instructions** - update and include at Top of Manage Stations Page?

**Comment by denislepage** [comment]
I can definitely see a value in having a list of receivers like we have now. The key question is to what purpose(s). We identified one today (pick a receiver for your new deployment), but that is happening on a different place, not the main station page.

I would be careful not to try and force this square peg into the round hole of the stations page if it doesn’t work. The station page is build for managing stations (duh), and the map/table on the left allows you to find existing stations based on different criteria (geographic, name or based on the receiver ID last used). If a particular receiver isn’t actively being used for instance, it won’t show up in that interface, at least not until you get to the point where you want to use it for a new deployment on the page we discussed today.

If we add that list as a tab on the station page, and we decide that all receivers should be shown there (e.g. including those not attached to a current deployment), 1) how is that list displayed, and 2) what functionalities will be tied to each row? How will cases where you have the same receiver attached to multiple stations being shown? Different rows? In summary, I’m not sure how the new table would differ from the table of stations we already have.

I think there is a need for users to see a list of receiver units like we had before, but I am not entirely sure of the different things this list might be used for, so I think we should clarify that. One such example would be to list all stations and/or deployments in which a specific receiver was used.

We could adjust the list of receiver page, such as this one, to include the concept of stations, first. You would still see a list of receivers (one row per receiver) and we could just add a column showing the number of stations, and/or keep the #deployments only.
https://motus.org/data/projectReceivers?n=1000&q=&f=1&o=0a1a2a3a&id=1

The receiver page (e.g. https://motus.org/data/receiver?id=2136) would then also need to be amended so you get a list of stations first, which would lead to the station page(s). I’m not sure whether we would allow going directly from that page to the individual deployments. Perhaps, but that might be too confusing and going against what we are trying to establish. That page could potentially also show instances where the receiver was used in other projects, without editing capabilities, etc.

Now, can the above 2 pages be integrated into a tab on the station page? Perhaps. The second page could be a modal window that shows up when you click on a receiver row, with a list of stations tied to that receiver.

Here’s a quick mock up (note that I also rearranged a bit the filter to be under the tabs, and moved the station block up as discussed. The second page would show what happens if you click on a receiver row. https://docs.google.com/presentation/d/1-2X6aixV0jqo_MY6gp1uQhJmT-cK-T2P/edit?usp=sharing&ouid=102988560087200266158&rtpof=true&sd=true. Perhaps the modal is only needed when you have more than 1 station for a receiver. Presumably nothing would happen if you click on a row where there are no stations for a receiver, or the modal would tell you that. We’d also want to add the count of stations in a column of the table.

Waddayathink? A functionality that we would potentially lose here is to look a receiver timeline, but I’m not sure the functionality would belong here, and that might need a dedicated page like the first link above.

**Comment by jsayers-bsc** [comment]
Bugs: 

- [ ] Clicking on "receiver" in the right hand side of the station brings up the same receiver despite the URL being correct (two images below, both accessed from the Stations page)

![image](https://user-images.githubusercontent.com/55001144/132695113-8561cb61-4de2-4a52-9fce-b909838a9c9b.png)

![image](https://user-images.githubusercontent.com/55001144/132695182-6d79723a-f6bd-4347-8384-46d2aea486e0.png)

**Comment by pmorrill** [comment]
I looks like the latest code is not deployed yet: there are a few updates that might fix the above issue.

**Comment by Hussell** [comment]
The updates have been deployed.

**Comment by pmorrill** [comment]
@jsayers-bsc : I am concerned with your statement:

On station management page, I get an java dev error when trying to create a new deployment for a station that is either active or inactive, but only if 'Use most recent receiver' is selected:

You should not be able to create a deployment on a station that is active. The interface should force you to terminate the active deployment first. Can you send me an example station so I can test?

**Comment by leberrigan** [comment]
@pmorrill, that one was me. It appears to be possible for all active stations which I have tried. There is no prompt for terminating the deployment.

Another issue:
- [ ] The text fragment key for confirming a receiver selection is not being matched with the one that currently exists. The only difference is the letter case. In addition, the existing text fragment includes a semicolon which we don't want to include on these buttons:
![image](https://user-images.githubusercontent.com/35931796/132713881-28fb57e4-cda9-45a8-b158-e641c95db198.png)

TX Keys: https://sandbox.motus.org/data/admin/tx_text?id=32
![image](https://user-images.githubusercontent.com/35931796/132713921-329e81f3-3415-4bde-b387-f70abab66108.png)

**Comment by pmorrill** [comment]
> @pmorrill, that one was me. It appears to be possible for all active stations which I have tried. There is no prompt for terminating the deployment.
>

Check here:

https://beta.motus.org/data/project/stations?projectID=1

@cjardine-bsc made a change in sandb ox only, to better handle deployment status calculation. Maybe it is not working? You shoujld always be prevented from creating a new config on a station with an active deployment. The sandbox seems to be ok....

**Comment by Hussell** [comment]
Update deployed.

**Comment by jsayers-bsc** [comment]
There is a "Delete Receiver" button in the deployment overview. Should that be "**Delete Deployment**"

![image](https://user-images.githubusercontent.com/55001144/132721733-79492a9a-bc57-49d4-96da-6521ba79ec64.png)

**Comment by jsayers-bsc** [comment]
- Ports are empty for many stations (even in edit mode)

![image](https://user-images.githubusercontent.com/55001144/132724351-d3ee6996-3acf-43db-a0ee-641988861da5.png)

- There is no option to modify dongle type. I imagine we want to have that as one of the properties that might be corrected in a deployment

**Comment by cjardine-bsc** [comment]
> 
> 
> > @pmorrill, that one was me. It appears to be possible for all active stations which I have tried. There is no prompt for terminating the deployment.
> 
> Check here:
> 
> https://beta.motus.org/data/project/stations?projectID=1
> 
> @cjardine-bsc made a change in sandbox only, to better handle deployment status calculation. Maybe it is not working? You should always be prevented from creating a new config on a station with an active deployment. The beta seems to be ok....

Let me know if there is a change needed for determining station status.

**Comment by pmorrill** [comment]
> 
> 
>     * [ ]  Ports are empty for many stations (even in edit mode)
> 
> 
> ![image](https://user-images.githubusercontent.com/55001144/132724351-d3ee6996-3acf-43db-a0ee-641988861da5.png)
> 

The jscript structure that carries dongle config info is missing it's port array on the beta server. It seems to be ok on the sandbox server. These are pulled from a lookup table lk_port_numbers, which looks to be populated. Can you find a similar issue on sandbox?

**Comment by pmorrill** [comment]
> 
> 
>     * [ ]  Modify properties in a station goes nowhere

?? Seems ok for me, now that the servers are responding

**Comment by jsayers-bsc** [comment]
> > * [ ]  Modify properties in a station goes nowhere
> > ```
> 
> ?? Seems ok for me, now that the servers are responding

Yup

**Comment by jsayers-bsc** [comment]
> 
> 
> > Also, is there a way to link from elsewhere directly to the page to edit a specific station?
> 
> Yes. By example:
> 
> https://beta.motus.org/data/project/stations?projectID=1&setStationId=197
> 
> The setStationId param triggers a redirect to dsplay the summary / manage tools

is this possible with Receivers as well?

**Comment by pmorrill** [comment]
yes - this already exists.

https://sandbox.motus.org/data/project/receivers?projectID=1&setRecvId=383

**Comment by pmorrill** [comment]
I posted a few updates this morning: some UX changes, but most importantly, a fix for the db exception we were seeing on Thursday. My local test db was missing a constraint that exists in Motus proper, so I was getting away with something that would never work in Motus live. Changes posted this morning should sort it so you can test in sandbox.

@denislepage  / @Hussell can re-deploy sandbox / beta at this point and we can move forward on the tests.

**Comment by Hussell** [comment]
We're back, after a 39 hour power outage. I've redeployed everything to sandbox and beta.

**Comment by jsayers-bsc** [comment]
> yes - this already exists.
> 
> https://sandbox.motus.org/data/project/receivers?projectID=1&setRecvId=383

This is nice, but upon page load the URL reverts back to just project ID. I can make a bookmarklet to add in and modify the &setRecvID=XXX but this does make sharing the URL more difficult (especially for collaborators trying to share a page they are on with us).

Can this be retained upon page load so it reflects the state of the page?

**Comment by jsayers-bsc** [comment]
Manage Receivers page:

- [ ] In table, Station should link to the station page and ID should link to the deployment page.

![image](https://user-images.githubusercontent.com/55001144/133646952-41d1b94b-b1ba-4baf-ae24-cd7168ed4994.png)

**Comment by jsayers-bsc** [comment]
Bug:

-[ ] Ports still don't seem to be showing up (this is actually a lookup table issue if I recall?)

![image](https://user-images.githubusercontent.com/55001144/133650477-352af01f-168a-4948-b120-b919708cd35b.png)

**Comment by cjardine-bsc** [comment]
I looked at the table content. The only difference was that port_config wasn't populated in lk_device_dongles in live. I've done so now. 

Let me know if that helps. If not, can you confirm it's working on sandbox? Thanks! 

Tables I checked:
lk_port_number
lk_dongle_frequencies
lk_port_config
lk_device_dongles
lk_device_type
lk_dongle_type

**Comment by pmorrill** [comment]
It seems to be working on sandbox, yes. But not on beta (I tried reloading the lookups static maps).

**Comment by cjardine-bsc** [comment]
Okay, if you can provide me a clue as to what table you're pulling the data from that might help. 

I'll run a DB compare now and try to find what I find. :)

**Comment by jsayers-bsc** [comment]
This is the list of links currently included in Manage Receivers. Something similar would be good in the tables in Manage Stations/Receivers (as described above)

![image](https://user-images.githubusercontent.com/55001144/133661342-4e83951e-5651-490d-8492-d5bee042e9a6.png)

**Comment by jsayers-bsc** [comment]
Still need access to tool to Delete Deployment

**Comment by pmorrill** [comment]
> 
> 
> Okay, if you can provide me a clue as to what table you're pulling the data from that might help.
> 
> I'll run a DB compare now and try to find what I find. :)

Here is the actual query:

select b.dongle_type, c.port, c.port_descr, c.sort from lk_receiver_type a
		inner join lk_device_dongles b
		on a.device_type = b.device_type
		inner join lk_port_numbers c on b.port_config = c.port_config
		where a.receiver_type = ? and b.dongle_type = ?
		order by c.sort

**Comment by pmorrill** [comment]
For example:

select b.dongle_type, c.port, c.port_descr, c.sort from lk_receiver_type a
							  inner join lk_device_dongles b
							  on a.device_type = b.device_type
							  inner join lk_port_numbers c on b.port_config = c.port_config
							  where a.receiver_type = 'Lotek' and b.dongle_type = 'Lotek Integrated'
							  order by c.sort

**Comment by cjardine-bsc** [comment]
query returning the same thing in sandbox and live.

**Comment by pmorrill** [comment]
> 
> 
> Still need access to tool to Delete Deployment

Under what conditions were you able to delete  a deployment in past? ie: as long as it had no start_date?

**Comment by jsayers-bsc** [comment]
This is on the Receiver Properties page (which is similar to our new receiver overview page). It's only available to admin.

![image](https://user-images.githubusercontent.com/55001144/133669758-db88ea44-7393-4c12-ba7c-16627beccef3.png)

**Comment by pmorrill** [comment]
Yikes - does it make sense to allow delete of deployments in all cases...when they have data processed for example? Does the database actually allow this?

**Comment by cjardine-bsc** [comment]
It does, using prDeleteSensorDeployment  it clears the deploy_id from all associated hit data. This might have been restricted to admin in the old version? 

Orphaned hits would trigger an error in the validation centre letting them know they have hits outside of a deployment.

**Comment by jsayers-bsc** [comment]
Developer error just now while trying to create a new deployment at a station where I just terminated the existing one... looks like due to a closed connection when the query was attempted.  SQL exception: com.microsoft.sqlserver.jdbc.SQLServerException: The connection is closed.

**Comment by jsayers-bsc** [comment]
![image](https://user-images.githubusercontent.com/55001144/134556724-1643cea0-fcdf-44eb-b1b7-247985260c42.png)

**Comment by pmorrill** [comment]
May need to restart the tomcat context. We have seen this a few times: motus-base does not get linked in on deploy of new code.

**Comment by jsayers-bsc** [comment]
Re Test stations we probably don't need to "most recent configuration section" as there may be many dozens of active deployments at any given time. Just the "create new" option would probably do. Active receiver deployments can be terminated by clicking and editing

**Comment by jsayers-bsc** [comment]
- Station map used to default to Active, but now defaults to All. Can that be restored?
- What's the likelihood of getting those shortcut links (deployment timeline, receiver timeline, deployment summary) in the deployment history table in Stations and Receivers page? (as discussed [above](https://github.com/MotusDev/Motus-TO-DO/issues/994#issuecomment-921113539)). In place of the Type field on Stations,

**Comment by leberrigan** [comment]
@pmorrill When trying clicking on 'terminate this deployment' while viewing a station, nothing happens and a js error is thrown: 
```
Uncaught SyntaxError: missing } after property list
stations:3:49note: { opened at line 2, column 5
```

**Comment by leberrigan** [comment]
While creating a new deployment and selecting a new receiver, it's possible to hit the 'back' button and re-select a new receiver over and over again, causing multiple concurrent deployments at the same station. Not only that, you can also pick the same receiver over and over again, causing there to be duplicate deployments for the same serial number.

I'm not sure if this is critical enough to address prior to release, but I believe this bug is likely to be triggered by regular use. For instance, it's likely possible to submit the form twice by accident, especially with a bad connection to the server.

**Comment by StuMackenzie** [comment]
**Properties**

Using: Malone University (# 5686)  - from https://beta.motus.org/data/project/stations?projectID=1

Status: - text fragment missing

When you select "Modify properties" - status is presented as a drop-down list - pending, prospective, available. 

There is also a prospective start date. 

This station is already active. Should it not be set to 'active' with a start date already populated, or those fields grayed out?

**Comment by Hussell** [comment]
> The deployment needs a rebuild, I think. The changes were checked in on Sep 29. But they are not showing in either beta or sandbox.

Try now. Everything should be up-to-date.

**Comment by leberrigan** [comment]
New bugs:
- **Beta** Adding a new station fails: 
![image](https://user-images.githubusercontent.com/35931796/137174886-74518932-24ea-4b34-a290-7bb0d02e5a82.png)

- **Sandbox and beta** Adding a new landowner fails page refreshes: 

- **Sandbox and beta** Menu items for _manage stations_ and _manage landowners_ directs you to main management page. It appears this is because these links do not include project ID in the URL, but that is not an issue for the other management pages.
  
- **Sandbox and beta** You can view station or landowner pages of projects that don't exist:
https://sandbox.motus.org/data/project/stations?projectID=4412

**Comment by jsayers-bsc** [comment]
Consolidating all outstanding previous comments/issues into one...

NOTE: the beta site has been unresponsive for the last couple hours so I haven't been able to test the bugs part to see how many are still relevant. 

**Bugs**
 - [ ] On [receiver management page](https://sandbox.motus.org/data/project/receivers?projectID=2), the key name for the header text fragment for the first column on the table has a typo: 
> Text fragment not found in database: (EN) **Recveiver_serno**
- [ ] When selecting a receiver for a new deployment, the text fragment key 'Deploy_receiver_to_station' is used for both the page header and submit button (the one used to 'select' the receiver for the new deployment and proceed to the next page). Can one of them be changed?
- [ ] On station management page, I get an java dev error when trying to create a new deployment for a station that is either active or inactive, but only if 'Use most recent receiver' is selected:
> java.lang.IllegalArgumentException: Unable to find 'column_name_unknown' in the table of error codes.
- [ ] Another dev error when confirming the selected receiver while creating a new deployment for a station that is either active or inactive, but only if "Use most recent antenna config" is selected.
- [ ] If "Use most recent antenna config" is not selected in the above scenario, the page just reloads.

**Changes**
 - [x] On [receiver management page](https://sandbox.motus.org/data/project/receivers?projectID=2), the right side panel has a table of deployments with links to deployments for the first column. I think it would be more intuitive to have these links on the second column with the deployment IDs.
   - I was going to say this would be helpful with links to both the station and the deployment in the table. Station name links to the station overview page and the deployment ID links to the deployment. - JBS
 - [ ] On the same page, the map button remains from previous interface. I think this may be better placed on the stations page, perhaps incorporated into the station map, but that may be time consuming. For now, the map button should can be placed next to it, or maybe as a tab.
